lua/lualine/components/diagnostics/init.lua

```diff
@@ -48,12 +50,25 @@ function M:init(options)
       '### diagnostics.sources\n\nno sources for diagnostics configured.\nPlease specify which diagnostics source you want lualine to use with `sources` option.\n'
     )
   end
+
+  -- Check if a language server reports working progress.
+  self.lsp_started = false
+  self.lsp_working = false
+  vim.lsp.handlers["$/progress"] = function(_, result)
+    self.lsp_started = true
+    if result.value.kind == "end" then
+      self.lsp_working = false
```

Comment:
`gopls` will emit a "begin" event when encountering an issue loading workspace, with no followup "end". In order to recover from bad actors, LSP progress handlers need to implement a debounced clear routine.